setup.py

```diff
@@ -1,48 +1,70 @@
 #! /usr/bin/env python
 
 import os
+import io 
 import sys
 import tempfile
 import shutil
 import builtins
 import textwrap
 
 import setuptools
+from configparser import ConfigParser
 from distutils.command.build_py import build_py
 from distutils.command.sdist import sdist
 from distutils.errors import CompileError, LinkError
 
+# Checking if Python installation is >=3.6
+if sys.version_info < (3, 6, 0):
 
-DISTNAME = 'scikit-image'
-DESCRIPTION = 'Image processing in Python'
-MAINTAINER = 'Stefan van der Walt'
-MAINTAINER_EMAIL = 'stefan@sun.ac.za'
-URL = 'https://scikit-image.org'
-LICENSE = 'Modified BSD'
-DOWNLOAD_URL = 'https://scikit-image.org/docs/stable/install.html'
-PROJECT_URLS = {
-    "Bug Tracker": 'https://github.com/scikit-image/scikit-image/issues',
-    "Documentation": 'https://scikit-image.org/docs/stable/',
-    "Source Code": 'https://github.com/scikit-image/scikit-image'
-}
-
-with open('README.md', encoding='utf-8') as f:
-    LONG_DESCRIPTION = f.read()
-
-if sys.version_info < (3, 6):
-
-    error = """Python {py} detected.
-
+    error = f"""Python {'.'.join([str(v) for v in sys.version_info[:3]])}
+detected. 
 scikit-image 0.16+ supports only Python 3.6 and above.
-
-For Python 2.7, please install the 0.14.x Long Term Support release using:
-
- $ pip install 'scikit-image<0.15'
-""".format(py='.'.join([str(v) for v in sys.version_info[:3]]))
+For Python 2.7, please install the 0.14.x Long Term Support release using: 
+$ pip install 'scikit-image<0.15'
+    """
 
     sys.stderr.write(error + "\n")
     sys.exit(1)
 
+
+# All settings are in configs.ini
+config = ConfigParser(delimiters=['='])
+config.read('setup.cfg')
+cfg = config['metadata']
+
+
+DISTNAME = cfg['name']
```

Comment:
I'm not sure you need to parse anything yourself.

https://github.com/pypa/setuptools/blob/master/setup.py

They don't do any of this parsing. I think setuptools will do this automatically for you if you don't specify the parameters.

Honestly. This kind of change is somewhat of a side-step. There are advantages to "declarative" build specifications, but I'm wondering if there is any measurable advantage to this change other than "keeping up with a shiny new tool from setuptools"

Reply:
Setuptools used a different way of parsing the configurations, of which I am unaware of. However, I have seen a number of popular packages use this method (which the PR is about). Honestly, there is no measurable advantage, but it helps neatly separate out the configuration variables.